### PR TITLE
Enhance persistent user store

### PR DESCRIPTION
**Pull request by CalamariDude**
Created loadUser for persistentDataStore to just load a single user from database.

### CODE REVIEW

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 1]
```diff
@@ -79,6 +79,66 @@ public PersistentDataStore() {
     return users;
   }
 
+  /**
+   * Loads unique User object using UUID from the Datastore service and returns it.
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public User loadUser(UUID id) throws PersistentDataStoreException {
```
Comment (hsha0) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: Indention here should be 4 spaces, you can select all codes and use shift-command-f for autoformat

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 2]
```diff
@@ -79,6 +79,66 @@ public PersistentDataStore() {
...
+
+    // Retrieve all users where uuid matches given UUID
+    Query query = new Query("chat-users");
+    query.setFilter(new Query.FilterPredicate("uuid", Query.FilterOperator.EQUAL, id.toString()));
+    PreparedQuery results = datastore.prepare(query);
+
+    //There should only be one user for each UUID
+    Entity entity = results.asSingleEntity();
```
Comment (hsha0) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: For line 90-96, indention should be 2 tab

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 3]
```diff
@@ -79,6 +79,66 @@ public PersistentDataStore() {
...
+          }
+  }
+
+  /**
+   * Loads unique User object using username from the Datastore service and returns it.
+   *
+   * @throws PersistentDataStoreException if an error was detected during the load from the
+   *     Datastore service
+   */
+  public User loadUser(String name) throws PersistentDataStoreException {
```
Comment (hsha0) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: Indention needs to be changed here and in the function

File: src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java [hunk 4]
```diff
@@ -70,6 +70,71 @@ public void testSaveAndLoadUsers() throws PersistentDataStoreException {
   }
 
   @Test
+  public void testSaveAndLoadSingleUserById() throws PersistentDataStoreException {
```
Comment (hsha0) [inline src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java]: change indention here and in the function

File: src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java [hunk 5]
```diff
@@ -70,6 +70,71 @@ public void testSaveAndLoadUsers() throws PersistentDataStoreException {
...
+
+        Assert.assertEquals(idTwo, resultUserTwo.getId());
+        Assert.assertEquals(nameTwo, resultUserTwo.getName());
+        Assert.assertEquals("test_password", resultUserTwo.getPassword());
+        Assert.assertEquals(creationTwo, resultUserTwo.getCreationTime());
+    }
+
+
+    @Test
   public void testSaveAndLoadConversations() throws PersistentDataStoreException {
```
Comment (hsha0) [inline src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java]: also indention

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 6]
```diff
@@ -79,6 +79,66 @@ public PersistentDataStore() {
...
+    Query query = new Query("chat-users");
+    query.setFilter(new Query.FilterPredicate("uuid", Query.FilterOperator.EQUAL, id.toString()));
+    PreparedQuery results = datastore.prepare(query);
+
+    //There should only be one user for each UUID
+    Entity entity = results.asSingleEntity();
+          try {
+              UUID uuid = UUID.fromString((String) entity.getProperty("uuid"));
+              String userName = (String) entity.getProperty("username");
+              String password = (String)entity.getProperty("password");
```
Comment (hsha0) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: add space between (String) and entity, just for a common format

**Review by hsha0 — COMMENTED** [review]
LGTM, except indentions

**Comment by CalamariDude** [comment]
Had to change the indentation settings on my ide, it loves its 4 spaces : )

I fixed the changes, merging.

**Pull request merged**